public/js/src/module/controller-webform.js

```diff
@@ -704,10 +719,9 @@ function _setEventHandlers( survey ) {
     }
 
     if ( settings.offline ) {
-        document.addEventListener( events.XFormsValueChanged().type, () => {
-            // The delay works around an issue with drawing widgets, where the canvas
-            // capture is an empty image. https://github.com/enketo/enketo-express/issues/174
-            setTimeout( _autoSaveRecord, 500 );
+        document.addEventListener( events.XFormsValueChanged().type, async () => {
+            await autoSavePromise;
+            await _autoSaveRecord();
```

Comment:
About `await _autoSaveRecord()`: since nothing follows it in the event listener, and a new event can execute the listener again before the previous one returns, does it do something substantive? Or is it here to remove a possible pitfall in case someone later adds more code to the listener?

Reply:
This is actually just a code style preference. I prefer to explicitly `await` when calling `async`/`Promise`-returning functions to make their asynchrony clear at the call site. Some people disagree with this, and it's not clear whether either is more idiomatic, but when in doubt I tend to err on the side of "explicit is better than implicit".